### PR TITLE
cython 3.0.0

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- update to cython 3.0.0

https://github.com/cython/cython/tree/3.0.0
https://github.com/cython/cython/blob/3.0.0/CHANGES.rst

### CODE REVIEW

**Review by ryanskeith — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -16,11 +16,12 @@ build:
     - cython = Cython.Compiler.Main:setuptools_main
     - cythonize = Cython.Build.Cythonize:main
     - cygdb = Cython.Debugger.Cygdb:main
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
```
Duplication

**Review by skupr-anaconda — COMMENTED** [review]
Is this v3.0.0 backward compatible for packages that implicitly relies on cython <3.0.0?
Should we identify and test them?

**Comment by cbouss** [comment]
> 

cython is a build dep that generates C++ code that gets compiled.
It should not affect packages runtime. It is sometimes declared as a run dep, but usually by mistake.

It is possible that we will have to restrict cython on new builds or rebuilds of feedstocks. But that should be taken care of when updating these feedstocks.

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**